syntaxes/vscode-autoit-output.tmLanguage.json

```diff
@@ -3,116 +3,167 @@
-        "2": {"name": "vscode-autoit-output-green"}
-      }
-    },
-    {
-      "match": "^(#[0-9]+: | {4,})?([-<].*)",
-      "captures":{
-        "1": {"name": "vscode-autoit-output-process-id"},
-        "2": {"name": "vscode-autoit-output-orange"}
-      }
-    },
-    {
-      "match": "^(#[0-9]+: | {4,})?([!].*)",
-      "captures":{
-        "1": {"name": "vscode-autoit-output-process-id"},
-        "2": {"name": "vscode-autoit-output-bold-red"}
-      }
-    },
-    {
-      "match": "^(#[0-9]+: | {4,})?(.*:([0-9]+):)",
-      "captures":{
-        "1": {"name": "vscode-autoit-output-process-id"},
-        "2": {"name": "vscode-autoit-output-purple"},
-        "3": {"name": "vscode-autoit-output-purple"}
-      }
-    },
-    {
-      "match": "^(#[0-9]+: | {4,})?(.*\\son line\\s([0-9]*).*)",
-      "captures":{
-        "1": {"name": "vscode-autoit-output-process-id"},
-        "2": {"name": "vscode-autoit-output-red"}
-      }
-    },
-    {
-      "match": "^(#[0-9]+: | {4,})?(@@ Debug\\(\\d+\\) :)",
-      "captures":{
-        "1": {"name": "vscode-autoit-output-process-id"},
-        "2": {"name": "vscode-autoit-output-debug"}
-      }
-    },
-    {
-      "match": "^(#[0-9]+: | {4,})?(.*\\(\\d+\\).*)",
-      "captures":{
-        "1": {"name": "vscode-autoit-output-process-id"},
-        "2": {"name": "vscode-autoit-output-olive"}
-      }
-    },
-    {
-      "match": "^(#[0-9]+: | {4,})?(In file included from\\s.*)",
-      "captures":{
-        "1": {"name": "vscode-autoit-output-process-id"},
-        "2": {"name": "vscode-autoit-output-purple"}
-      }
-    },
-    {
-      "match": "^(#[0-9]+: | {4,})?(line(?:\\s|.)*\\scolumn(?:\\s|.)*)",
-      "captures":{
-        "1": {"name": "vscode-autoit-output-process-id"},
-        "2": {"name": "vscode-autoit-output-red"}
-      }
-    },
-    {
-      "match": "^(#[0-9]+: | {4,})?(.*\\t([0-9]+)\\s.*)",
-      "captures":{
-        "1": {"name": "vscode-autoit-output-process-id"},
-        "2": {"name": "vscode-autoit-output-pink"}
-      }
-    },
-    {
-      "match": "^#[0-9]+: | {4,}",
-      "name": "vscode-autoit-output-process-id"
+    "general": {
+      "patterns": [
+        {
+          "begin": "\\G(Starting process (#\\d+))",
+          "end": "((.+) \\[PID (\\d+|n\/a)\\])",
+          "beginCaptures": {
+            "1": { "name": "vscode-autoit-output-process" },
+            "2": { "name": "vscode-autoit-output-process-id" }
+          },
+          "endCaptures": {
+            "1": { "name": "vscode-autoit-output-process" },
+            "2": { "name": "vscode-autoit-output-process-file" },
+            "3": { "name": "vscode-autoit-output-process-pid" }
+          }
+        },
+        {
+          "comment": "exit with no error code (0 or -1)",
+          "match": "\\G>Exit code (?:0|-1) .*",
+          "name": "vscode-autoit-output-EXIT"
+        },
+        {
+          "match": "\\G>Exit code 1 .*",
```

Comment:
`(\\G|^)`

Reply:
the`↑` doesn't match anything in nested patterns...

Reply:
your nested pattern doesn't go over a newline for it to match
it ends before it

Reply:
yes, that's why it was replaced with `\\G`, but you commented with `(\\G|^)`, as if `^` would do anything...

Reply:
^ will only work if something pushes the end rule out
currently you have nothing that does that

Reply:
is there an example file I can look at?
to see what possible solutions there are

Reply:
This is the text I'm testing it against, and image of SciTe editor output that I'm trying make this extension to match to
[output.zip](https://github.com/loganch/AutoIt-VSCode/files/9738193/output.zip)
So far, it's working 100%